src/ui/Modal/Modal.tsx

```diff
@@ -33,36 +37,73 @@ const Modal: FC<ModalProps> = ({
   const modalId = useId(id);
   const modalTitleId = `modal-title-${modalId}`;
   return (
-    <Dialog
-      id={modalId}
-      aria-labelledby={modalTitleId}
-      css={modalDialogCss(size)}
-      isOpen={show}
-      onDismiss={onClose}
-      {...props}
-    >
-      {header && (
-        <ModalHeader>
-          <ButtonGhost
-            icon
-            css={closeModalBtnCss(closeButtonPosition)}
-            onClick={onClose}
+    <AnimatePresence>
+      {/* Required to aniamte exit */}
+      {show && (
+        <AnimatedReachDialogOverlay
+          onDismiss={onClose}
+          transition={{ duration: animationDuration }}
+          variants={overlayAnimationVariants}
+          initial="hidden"
+          animate="visible"
+          exit="hidden"
+        >
+          <AnimatedReachDialogContent
+            id={modalId}
+            aria-labelledby={modalTitleId}
+            css={modalDialogCss(size)}
+            {...props}
+            transition={{ duration: animationDuration }}
```

Comment:
Nit: should use `spring` animation, it's more realistic. Using `duration` set animation to `tween`.